### PR TITLE
naming advice around `inj` and `ext`

### PR DESCRIPTION
**Pull request by semorrison**

### CODE REVIEW

File: templates/contribute/naming.md [hunk 1]
```diff
@@ -259,6 +259,40 @@ check le_of_mul_le_mul_left
...
 ```
 
+## Naming of structural lemmas ##
+
+We are trying to standardize certain naming patterns for structural lemmas.
+At present these are not uniform across mathlib.
+
+### Extensionality ###
+
+A lemma of the form `(∀ a, a ∈ s₁ ↔ a ∈ s₂) → s₁ = s₂` should be named `.ext`,
```
Comment (urkud) [inline templates/contribute/naming.md:269]: We also have `∀ x, f x = g x` as a common assumption for `@[ext]`.

File: templates/contribute/naming.md [hunk 2]
```diff
@@ -259,6 +259,40 @@ check le_of_mul_le_mul_left
...
+and labelled with the `@[ext]` attribute.
+Often this type of lemma can be generated automatically by putting the
+`@[ext]` attribute on a structure.
```
Comment (urkud) [inline templates/contribute/naming.md:272]: AFAIK `@[ext]` uses projections while quite often `ext` lemmas should use coercions.

Comment (semorrison) [reply templates/contribute/naming.md:272]: Good point. I will make this explicit.

Comment (semorrison) [inline templates/contribute/naming.md:272]:
```suggestion
Often this type of lemma can be generated automatically by putting the
`@[ext]` attribute on a structure.
(However an automatically generated lemma will always be written in terms
of the structure projections, and often there is a better statement,
e.g. using coercions, that should be written by hand then marked with `@[ext]`.)

```

File: templates/contribute/naming.md (hunk 1 above)
Comment (semorrison) [reply templates/contribute/naming.md:269]:
```suggestion
A lemma of the form `(∀ x, f x = g x) → f = g` should be named `.ext`,
```

File: templates/contribute/naming.md [hunk 3]
```diff
@@ -259,6 +259,40 @@ check le_of_mul_le_mul_left
...
+
+A lemma of the form `s₁ = s₂ ↔ ∀ a, a ∈ s₁ ↔ a ∈ s₂` should be named `.ext_iff`.
```
Comment (semorrison) [inline templates/contribute/naming.md:274]:
```suggestion
A lemma of the form `f = g ↔ ∀ x, f x = g x` should be named `.ext_iff`.
```

**Comment by jcommelin** [comment]
I think it's good to codify what has been an implicit naming practice already. I'm fine with merging this. @bryangingechen @robertylewis @digama0 any comments?

**Comment by bryangingechen** [comment]
I'm happy with merging this too.

**Comment by robertylewis** [comment]
I'm not entirely up on the naming conventions but this looks fine to me. I don't think the trailing `###`s are necessary.

**Comment by bryangingechen** [comment]
> I don't think the trailing `###`s are necessary.

They're just there for consistency with the rest of the file, I think. We can get rid of them in a cleanup pass later on. See #62 as well as the duplicated variable names table in style.md.

**Pull request merged**